### ISSUE TITLE
[idea] hooks

### ISSUE THREAD

**Issue by dpastoor** [AstraZeneca/blueprint#9]
thinking about the idea of having a catch all `add_hook` that would allow user templates to be easily managed.

eg imagine you wanted to add a field at the top of `$PK`, you could update the template file

```
...
$PK
{{{hooks:pk}}}
...
```

then in the blueprint

```
bp %>% add_hooks(pk = "<whatever code you'd like to add>")
```

@HelenaEdlund does this (somewhat) make sense - eg rather than needing to do string replacement later, though that is still possible, provide a better mechanism for being able to define how to pass strings/lists to the templating engine to run.

**Comment by HelenaEdlund** [comment]
@dpastoor Yes, this makes sense to me. I think we talked about this concept some time ago. Would be great to be able to do this in an efficient way instead of string replacements. Just adding a calculation or a lumping variable of groups in your data or whatever. 

Using it the last days I also found that I want to wait as long as I can and render only when writing it to file so that I can keep piping on things to create a new run based on a "parent" run so to say. And if I need to do it at the rendered version that makes it less efficient.

**Comment by dpastoor** [comment]
prelim implementation https://github.com/AstraZeneca/blueprint/pull/15 going to keep this issue open as more functionality gets scoped

- [ ] show what hooks are available for a template
- [ ] merging hooks - eg add another line below on the same hook
- [ ] multi line hooks (currently supports single string that needs to be concat'd together
